clusterversion: create 25.1 version

RaduBerinde opened the pull request:
#### clusterversion: create 25.1 version

This change bumps the version on master to 25.1 without advancing the
previous release to 24.3 (the latter can only be done once we have a
published RC, or upgrade tests will fail). To allow this decoupling, a
few minor tweaks were needed (to `SupportedPreviousReleases` and to
the tool that generates the releases file.

Also made some minor improvements to the runbook and
sql-bootstrap-data. I will do a more thorough change to the runbook
that separates this step.

- [x] Add version key constant for new release, equal to `Latest`
- [x] Add start version
- [x] Update `pkg/build/version.txt` to the new version
- [x] Update the `scplan` rules in `pkg/sql/schemachanger/scplan/internal/rules` (already done)
- [x] Update releases file

Epic: REL-1322

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/134750)
<!-- Reviewable:end -->

RaduBerinde committed:
clusterversion: create 25.1 version

This change bumps the version on master to 25.1 without advancing the
previous release to 24.3 (the latter can only be done once we have a
published RC, or upgrade tests will fail). To allow this decoupling, a
few minor tweaks were needed (to `SupportedPreviousReleases` and to
the tool that generates the releases file.

Also made some minor improvements to the runbook and
sql-bootstrap-data. I will do a more thorough change to the runbook
that separates this step.

- [x] Add version key constant for new release, equal to `Latest`
- [x] Add start version
- [x] Update `pkg/build/version.txt` to the new version
- [x] Update the `scplan` rules in `pkg/sql/schemachanger/scplan/internal/rules` (already done)
- [x] Update releases file

Epic: REL-1322

dt commented on pkg/cmd/release/update_releases.go:
```diff
@@ -334,12 +334,13 @@ func generateRepositoriesFile(versions ...string) error {
 			}
 			if resp.StatusCode != http.StatusOK {
-				return fmt.Errorf("unexpected status code %d when downloading %s", resp.StatusCode, url)
+				return fmt.Errorf("unexpected status code %d when downloading %s; consider turning off NetSkope", resp.StatusCode, url)
```
I believe we avoid mentioning these sorts of specifics. Maybe say "disabling any web traffic monitoring tools"

RaduBerinde replied:
Undid this change, it looks like it was a transient issue on my machine

RaduBerinde committed:
schemachanger: switch check to V25_1

RaduBerinde commented:
CC @spilchen for the second commit.

celiala commented on pkg/clusterversion/README.md:
```diff
@@ -183,7 +183,10 @@ avoid large changes on `master` which might cause merge conflicts for backports.
 
 - [ ] Update `PreviousRelease` constant
 
```
thank you for keeping the runbook up-to-date!

celiala commented on pkg/cmd/release/update_releases.go:
```diff
@@ -126,10 +126,10 @@ func processReleaseData(data []Release) map[string]release.Series {
+		// For the purposes of the cockroach_releases file, we are only interested
+		// in rc pre-releases, as we do not support upgrades from alpha or beta
+		// releases.
+		if pre := v.PreRelease(); pre != "" && !strings.HasPrefix(pre, "rc") {
```
nice fix/solution 👍🏼

celiala commented on pkg/cmd/release/update_releases_test.go:
```diff
@@ -39,10 +39,6 @@ func Test_processReleaseData(t *testing.T) {
 			Predecessor: "22.2",
 			Withdrawn:   []string{"23.1.0"},
 		},
-		"23.2": {
```
to confirm, this removal/cleanup is more related to advancing `MinSupported` to 24.3 (than to creating 24.3 and 25.1 start version gates, right?)

(mostly to clarify for when we refer to this for the next release cycle/task)

RaduBerinde replied:
This is just a test for the updated releases code, which we updated to ignore beta builds. This test runs against fixed data, the versions should be considered fictitious.

celiala commented on pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion_test.go:
```diff
@@ -244,7 +245,7 @@ func Test_choosePreviousReleases(t *testing.T) {
 			numUpgrades:       3,
 			enableSkipVersion: true,
-			expectedReleases:  []string{"23.2.4", "24.1.1", "24.2.2"},
+			expectedReleases:  []string{"23.1.17", "23.2.4", "24.1.1"},
```
x-referencing [note from MinSupport bump comment](https://github.com/cockroachdb/cockroach/pull/134340#pullrequestreview-2419388116) (for when we do this task for the next cycle), that this change is a result of the randomized sequence of versions that the test decides on with that seed.

RaduBerinde replied:
Right, and it's all based on the testPredecessorMapping (not the actual release data)

celiala commented on pkg/testutils/release/cockroach_releases.yaml:
```diff
@@ -29,6 +29,5 @@
   withdrawn:
   - 24.2.1
   predecessor: "24.1"
-"24.3":
```
we should keep 24.3, yes?

celiala reviewed (commented):
I had one question around keeping 24.3 (instead of replacing) for `pkg/testutils/release/cockroach_releases.yaml`. 

everything else made sense to me.

celiala reviewed (approved):
LGTM - thank you!

RaduBerinde commented:
TFTRs!

bors r+
